components/suggestion/switch_channel_provider.jsx

```diff
@@ -763,11 +761,11 @@ export default class SwitchChannelProvider extends Provider {
                 );
                 wrappedChannel = {...wrappedChannel, ...userWrappedChannel};
             }
-            const unread = allUnreadChannelIdsSet.has(channel.id);
+            const unread = allUnreadChannelIdsSet.has(channel.id) && !isChannelMuted(member);
             if (unread) {
                 wrappedChannel.unread = true;
             }
-            wrappedChannel.type = channelType;
+            wrappedChannel.type = isChannelMuted(member) ? Constants.MENTION_RECENT_CHANNELS : channelType;
```

Comment:
@sinansonmez I think we don’t need this change, Or maybe I am not able to understand why we want to change channel type if the channel is Muted.

Reply:
This is actually pushing down the muted channel to `Recent` section of the channels list. If I just change `unread` it just changes the format (i.e. making it unbold)

Reply:
@sinansonmez I would suggest filtering the muted channels with a mention here instead of what you have done above
https://github.com/mattermost/mattermost-webapp/blob/dfc7c4b2c096d7e422c70e4c111f3cb09e166876/components/suggestion/switch_channel_provider.jsx#L662

@esethna I tested it again, muted channels have the lowest priority, i.e they will only show if the user has less than five unmuted unread Channels.

Reply:
Updated as suggested, thanks @AshishDhama. I also agree this way much makes more sense.